op-batcher/batcher/channel_manager.go

```diff
@@ -199,14 +202,23 @@ func (s *channelManager) ensureChannelWithSpace(l1Head eth.BlockID) error {
 	}
 
 	pc, err := newChannel(s.log, s.metr, s.cfg, s.rollupCfg)
+
 	if err != nil {
 		return fmt.Errorf("creating new channel: %w", err)
 	}
+
+	if s.l1OriginLastClosedChannel.Number > 0 {
+		// Sets the timeout of the channel to be the latest safe head + max channel timeout
+		pc.RegisterL1Block(s.l1OriginLastClosedChannel.Number)
+	}
```

Comment:
We're now changing the duration tracking logic to set the channel duration timeout at channel creation. I think it's worth doing some cleaning up of what the channel and builder methods do to stay semantically clear (and avoid closing the channel at creation).

`channel.RegisterL1Block` is called whenever a new l1 head is seen. It's currently doing two things (via its `ChannelBuilder`)
  * `c.updateDurationTimeout(l1BlockNum)` -> maybe we can remove this, because we're now setting up the channel duration timeout at channel creation? I see that we would only want to remove it if we guarantee that we set it at channel creation.
  * `c.checkTimeout(l1BlockNum)` -> It may make sense to keep this as the only action. Note that this also marks timed-out channels as full.

So I suggest instead of calling `pc.RegisterL1Block` here, which might also close the channel immediately, we expose `updateDurationTimeout` on the `ChannelBuilder` and `channel` and call it here instead. We could also rename the parameter of `updateDurationTimeout` to something like `latestL1OriginBlockNum`.

```suggestion
	pc, err := newChannel(s.log, s.metr, s.cfg, s.rollupCfg)

	if err != nil {
		return fmt.Errorf("creating new channel: %w", err)
	}

	if s.l1OriginLastClosedChannel.Number > 0 {
		// Sets the timeout of the channel to be the latest safe head + max channel timeout
		pc.UpdateDurationTimeout(s.l1OriginLastClosedChannel.Number)
	}
```

We could even pass the l1Origin block timestamp as a parameter to `newChannel`, forcing the `latestL1Origin` to be determined at channel creation. But I'm not sure, is it possible at all that `s.l1OriginLastClosedChannel.Number` is equal to zero here?

Reply:
I've updated this to pass it through as a param on creation (I think this makes the intention much clearer). 

In theory the value should never be zero, as we default to the previous safe head (or L1 genesis block) and only update when it's [greater than](https://github.com/ethereum-optimism/optimism/pull/9769/files#diff-6c4365d39a27213e51b037de84f57824e828cceb6ba728b3d2eed0a4ea449bb5R296-R299) that. So even if there was a zero block channel or something peculiar, it shouldn't get set to zero.